### DISCUSSION TITLE
`scr` typefaces function to script letters

### DISCUSSION THREAD

**Discussion by SnO2WMaN** [typst/typst#1961] [Ideas]
Typst currently has `cal` as a calligraphic font (https://typst.app/docs/reference/math/variants/#cal), but it would be helpful to add the function `scr` as a script font. (See, for example, `\mathscr` at https://katex.org/docs/supported#style-color-size-and-font.) I need some of these alphabets in mathematical logic (when porting some my documents from written in TeX), but another example is the style used in the Hamiltonian of physics.

current `cal` in Typst
![image](https://github.com/typst/typst/assets/15155608/b06c369a-60c6-47fa-b351-ef997cbe0ba0)

`mathscr` in KaTeX
![image](https://github.com/typst/typst/assets/15155608/ed512dce-a252-4a71-bc80-0201a6785bd0)

My concern is that at least [source code](https://github.com/typst/typst/blob/3ab179ebd3157107bd650e3abc4e644fc183f79b/crates/typst-library/src/math/style.rs), `cal` is assigned in Unicode to the "Mathematical Script" series, so I think the only way to add `scr` would be "Mathmetical Bold Script", but have some doubts about whether that is reasonable or the point that there is also the issue of fonts.

**Comment by bluebear94** [comment]
Libertinus Math has the Mathematical Bold Script series as a bold version of Mathematical Script.

FreeSerif has the Mathematical Script series looking like `scr` and the Mathematical Bold Script series looking like `cal`.

**Comment by Enivex** [comment]
Unicode simply only has one range of calligraphic characters, and it's up to the font whether they should look like what LaTeX calls `\mathcal` or `\mathscr`.

It's therefore not possible for typst to provide a reliable `scr` command. Though you can make your own. Either by assigning `scr` to a different font with the look you want, or to the same font with different feature flags. Some fonts, like `New Computer Modern Math` and `Stix Two Math` use such flags to let the user pick a style for those codepoints.

**Reply by Trebor-Huang** [reply]
There seems to be a unicode variant selector for Chancery hand (`mathcal`) and Round hand (`mathscr`) on these characters.

**Reply by Enivex** [reply]
> There seems to be a unicode variant selector for Chancery hand (`mathcal`) and Round hand (`mathscr`) on these characters.

Yes, that is a brand new thing. There's an open PR regarding this at this very moment.
